### PR TITLE
Add Support for Azure Blob Storage Transfer

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/8695

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
storage-transfer: added support for `azure_blob_storage_data_source` to `google_storage_transfer_job`
```

### CODE REVIEW

File: mmv1/products/bigquerydatatransfer/api.yaml [hunk 1]
```diff
@@ -39,7 +39,7 @@ objects:
     references: !ruby/object:Api::Resource::ReferenceLinks
       guides:
         "Official Documentation": "https://cloud.google.com/bigquery/docs/reference/datatransfer/rest/"
-      api: "https://cloud.google.com/bigquery/docs/reference/datatransfer/rest/v1/projects.locations.transferConfigs/create"
+      api: "https://cloud.google.com/bigquery-transfer/docs/reference/datatransfer/rest/v1/projects.locations.transferConfigs"
```
Comment (melinath) [inline mmv1/products/bigquerydatatransfer/api.yaml:42]: This seems to be an unrelated change; could you remove it from this PR?

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go [hunk 2]
```diff
@@ -370,6 +379,44 @@ func httpDataSchema() *schema.Resource {
...
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: `The container to transfer from the Azure Storage account.`,
+			},
+			"path": {
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: `Root path to transfer objects. Must be an empty string or full path name that ends with a '/'. This field is treated as an object prefix. As such, it should generally not begin with a '/'.`,
+			},
+			"azure_credentials": {
```
Comment (melinath) [inline mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:400]: It looks like this is input-only and should possibly be ForceNew: https://cloud.google.com/storage-transfer/docs/reference/rest/v1/TransferSpec#AzureBlobStorageData

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go [hunk 3]
```diff
@@ -370,6 +379,44 @@ func httpDataSchema() *schema.Resource {
...
+				Type:        schema.TypeString,
+				Description: `The name of the Azure Storage account.`,
+			},
+			"container": {
+				Required:    true,
+				Type:        schema.TypeString,
+				Description: `The container to transfer from the Azure Storage account.`,
+			},
+			"path": {
+				Required:    true,
```
Comment (melinath) [inline mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:396]: Can it be required and still be an empty string?

**Review by melinath — CHANGES_REQUESTED** [review]
This seems reasonable - just a couple requests / questions. Thanks for the PR!

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go (hunk 3 above)
Comment (upodroid) [reply mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:396]: No, changed it to Optional + Computed

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go (hunk 2 above)
Comment (upodroid) [reply mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:400]: I'm not sure, the AWS credentials aren't ForceNew either. I'm assuming it behaves the same way.

File: mmv1/products/bigquerydatatransfer/api.yaml (hunk 1 above)
Comment (upodroid) [reply mmv1/products/bigquerydatatransfer/api.yaml:42]: It is a doc fix. Google updated the url + the method shouldn't be in the url.

Comment (melinath) [reply mmv1/products/bigquerydatatransfer/api.yaml:42]: thanks for the docs fix; since it's an unrelated change, it would be great to have it in a separate PR.

File: mmv1/third_party/terraform/resources/resource_storage_transfer_job.go (hunk 2 above)
Comment (melinath) [reply mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:401]: hmm... the docs also say that AWS creds are input-only. I think that's probably a bug waiting to happen; we should trust the docs and make this ForceNew.

Comment (upodroid) [reply mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:401]: The documentation is stale. I got hold of an Azure account for testing and updating the SAS Token doesn't recreate the resource.

Unfortunately, Storage Transfer doesn't log anything in Stackdriver and I created it in the console.


```
C02S62ZFFVH6:Git mahamed$ curl -H "Authorization: Bearer $(gcloud auth application-default print-access-token)" https://storagetransfer.googleapis.com/v1/transferJobs/17436582952808956420?project_id=550924169191
{
  "name": "transferJobs/17436582952808956420",
  "description": "Provider Development",
  "projectId": "REDACTED",
  "transferSpec": {
    "gcsDataSink": {
      "bucketName": "videndum-tmp"
    },
    "objectConditions": {},
    "transferOptions": {},
    "azureBlobStorageDataSource": {
      "storageAccount": "mahamedgcstransfer",
      "container": "dev"
    }
  },
  "schedule": {
    "scheduleStartDate": {
      "year": 2021,
      "month": 5,
      "day": 19
    },
    "scheduleEndDate": {
      "year": 2021,
      "month": 5,
      "day": 19
    },
    "endTimeOfDay": {}
  },
  "status": "ENABLED",
  "creationTime": "2021-05-19T14:29:58.948051753Z",
  "lastModificationTime": "2021-05-19T14:50:30.018660834Z",
  "latestOperationName": "transferOperations/transferJobs-17436582952808956420--7984150713314667712"
}
```

Comment (upodroid) [reply mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:401]: image

Comment (melinath) [reply mmv1/third_party/terraform/resources/resource_storage_transfer_job.go:401]: okay great, thanks for testing this! Sorry for the delayed response; just saw the notification.

**Comment by melinath** [comment]
/gcbrun; the last failure was a transient agent issue. @upodroid could you make the last requested change? Then I can approve and merge.

File: mmv1/products/bigquerydatatransfer/api.yaml (hunk 1 above)
Comment (upodroid) [reply mmv1/products/bigquerydatatransfer/api.yaml]: Reverted

**Review by melinath — APPROVED** [review]
